// fs-lix/index.js
var fs = require('fs');
var lix = require('_lix2js');

function asyncJsToLix(cb)
{
	return function () {
		var args = arguments;
		args[args.length - 1] = lix.func(args[args.length - 1]);
		return function () {
			cb.apply(null, args);
		}
	}
}

module.exports = function () {
	return {
		RenameAsync: asyncJsToLix(fs.rename),
		FtruncateAsync: asyncJsToLix(fs.ftruncate),
		TruncateAsync: asyncJsToLix(fs.truncate),
		ChownAsync: asyncJsToLix(fs.chown),
		LchownAsync: asyncJsToLix(fs.lchown),
		ChmodAsync: asyncJsToLix(fs.chmod),
		FchmodAsync: asyncJsToLix(fs.fchmod),
		LchmodAsync: asyncJsToLix(fs.lchmod),
		StatAsync: asyncJsToLix(fs.stat),
		FstatAsync: asyncJsToLix(fs.fstat),
		LstatAsync: asyncJsToLix(fs.lstat),
		LinkAsync: asyncJsToLix(fs.link),
		SymlinkAsync: asyncJsToLix(fs.symlink),
		ReadlinkAsync: asyncJsToLix(fs.readlink),
		ReadpathAsync: asyncJsToLix(fs.realpath),
		UnlinkAsync: asyncJsToLix(fs.unlink),
		RmdirAsync: asyncJsToLix(fs.rmdir),
		MkdirAsync: asyncJsToLix(fs.mkdir),
		ReaddirAsync: asyncJsToLix(fs.readdir),
		CloseAsync: asyncJsToLix(fs.close),
		OpenAsync: asyncJsToLix(fs.open),
		UtimesAsync: asyncJsToLix(fs.utimes),
		FutimesAsync: asyncJsToLix(fs.futimes),
		FsyncAsync: asyncJsToLix(fs.fsync),
		WriteAsync: asyncJsToLix(fs.write),
		ReadAsync: asyncJsToLix(fs.read),
		ReadFileAsync: asyncJsToLix(fs.readFile),
		WriteFileAsync: asyncJsToLix(fs.writeFile),
		AppendFileAsync: asyncJsToLix(fs.appendFile),
		WatchFileAsync: asyncJsToLix(fs.watchFile),
//		UnwatchFileAsync: asyncJsToLix(fs.unwatchFile),
//		WatchAsync: asyncJsToLix(fs.watch),
		ExistsAsync: asyncJsToLix(fs.exists),
		ExistsAsync: asyncJsToLix(fs.exists),
	}
}
